plugins/k8saudit-aks/pkg/k8sauditaks/k8sauditaks.go

```diff
@@ -0,0 +1,261 @@
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/plugins/source"
+	"github.com/falcosecurity/plugins/plugins/k8saudit/pkg/k8saudit"
+	falcoeventhub "github.com/falcosecurity/plugins/shared/go/azure/eventhub"
+	"github.com/invopop/jsonschema"
+	"golang.org/x/time/rate"
+)
+
+const pluginName = "k8saudit-aks"
+const regExpAuditID = `"auditID":[ a-z0-9-"]+`
+
+var regExpCAuditID *regexp.Regexp
+
+type Plugin struct {
+	k8saudit.Plugin
+	Logger *log.Logger
+	Config PluginConfig
+}
+
+type PluginConfig struct {
+	EventHubNamespaceConnectionString string `json:"event_hub_namespace_connection_string" jsonschema:"title=event_hub_namespace_connection_string,description=The connection string of the EventHub Namespace to read from"`
+	EventHubName                      string `json:"event_hub_name" jsonschema:"title=event_hub_name,description=The name of the EventHub to read from"`
+	BlobStorageConnectionString       string `json:"blob_storage_connection_string" jsonschema:"title=blob_storage_connection_string,description=The connection string of the Blob Storage to use as checkpoint store"`
+	BlobStorageContainerName          string `json:"blob_storage_container_name" jsonschema:"title=blob_storage_container_name,description=The name of the Blob Storage container to use as checkpoint store"`
+	RateLimitEventsPerSecond          int    `json:"rate_limit_events_per_second" jsonschema:"title=rate_limit_events_per_second,description=The rate limit of events per second to read from EventHub"`
+	RateLimitBurst                    int    `json:"rate_limit_burst" jsonschema:"title=rate_limit_burst,description=The rate limit burst of events to read from EventHub"`
+}
+
+func (p *Plugin) Info() *plugins.Info {
+	return &plugins.Info{
+		ID:          21,
+		Name:        pluginName,
+		Description: "Read Kubernetes Audit Events for AKS from EventHub and use blob storage as checkpoint store",
+		Contact:     "github.com/falcosecurity/plugins",
+		Version:     "0.1.0",
+		EventSource: "k8s_audit",
+	}
+}
+
+// Reset sets the configuration to its default values
+func (p *PluginConfig) Reset() {
+	if i := os.Getenv("EVENTHUB_NAMESPACE_CONNECTION_STRING"); i != "" {
+		p.EventHubNamespaceConnectionString = i
+	}
+	if i := os.Getenv("EVENTHUB_NAME"); i != "" {
+		p.EventHubName = i
+	}
+	if i := os.Getenv("BLOB_STORAGE_CONNECTION_STRING"); i != "" {
+		p.BlobStorageConnectionString = i
+	}
+	if i := os.Getenv("BLOB_STORAGE_CONTAINER_NAME"); i != "" {
+		p.BlobStorageContainerName = i
+	}
+	if i := os.Getenv("RATE_LIMIT_EVENTS_PER_SECOND"); i != "" {
+		rateLimitEventsPerSecond, err := strconv.Atoi(i)
+		if err != nil {
+			return
+		}
+		p.RateLimitEventsPerSecond = rateLimitEventsPerSecond
+	} else {
+		p.RateLimitEventsPerSecond = 100
+	}
+	if i := os.Getenv("RATE_LIMIT_BURST"); i != "" {
+		rateLimitBurst, err := strconv.Atoi(i)
+		if err != nil {
+			return
+		}
+		p.RateLimitBurst = rateLimitBurst
+	} else {
+		p.RateLimitBurst = 200
+	}
+}
```

Comment:
I don't recommend using env vars to override the config.

Moreover, Falco is able to use env vars in the config (here's an [example](https://github.com/falcosecurity/falco/blob/master/falco.yaml#L409)) which is preferable.